### PR TITLE
Get/Set Object lock config APIs

### PR DESCRIPTION
**Pull request by prakashsvmx**
Get/Set Object locking
- `makeBucket(bucketName, region[, makeOpts , callback])`

- `setObjectLockConfig(bucketName, lockConfig [, callback])`

- `getObjectLockConfig(bucketName [, callback])`

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -451,6 +462,68 @@ minioClient.setBucketVersioning('bucketname',versioningConfig, function (err){
...
+
+__Example __
+Get object lock on a Bucket
+
+```js
+s3Client.getObjectLockConfig('my-bucketname', function (err, lockConfig){
+    if (err) {
+        return console.log(err)
+    }
+    console.log("Success",lockConfig)
```
Comment (ebozduman) [inline docs/API.md:524]:
```suggestion
    console.log(lockConfig)
```

File: examples/get-object-lock-config.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  accessKey: 'YOUR-ACCESSKEYID',
+  secretKey: 'YOUR-SECRETACCESSKEY'
+})
+
+
+s3Client.getObjectLockConfig('my-bucketname', function (err,lockConfig){
+  if (err) {
+    return console.log(err)
+  }
+  console.log("Success", lockConfig)
```
Comment (ebozduman) [inline examples/get-object-lock-config.js:33]:
```suggestion
  console.log(lockConfig)
```

File: examples/set-object-lock-config.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  console.log("Success")
+})
+
+//To reset/remove object lock config.
+s3Client.setObjectLockConfig('my-bucketname', {}, function (err){
+  if (err) {
+    return console.log(err)
+  }
+  console.log("Success")
+})
```
Comment (ebozduman) [inline examples/set-object-lock-config.js:42]: I think `delete` should be a separate example and needs to be removed from the `set/put` example.

**Review by ebozduman — CHANGES_REQUESTED** [review]

**Comment by kannappanr** [comment]
```
S3Error: Missing required header for this request: Content-MD5
    at Object.parseError (/home/kannappan/work/github.com/minio/minio-js/dist/main/xml-parsers.js:83:11)
    at /home/kannappan/work/github.com/minio/minio-js/dist/main/transformers.js:158:22
    at DestroyableTransform._flush (/home/kannappan/work/github.com/minio/minio-js/dist/main/transformers.js:82:10)
    at DestroyableTransform.prefinish (/home/kannappan/work/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_transform.js:129:10)
    at DestroyableTransform.emit (events.js:310:20)
    at prefinish (/home/kannappan/work/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:611:14)
    at finishMaybe (/home/kannappan/work/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:620:5)
    at endWritable (/home/kannappan/work/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:643:3)
    at DestroyableTransform.Writable.end (/home/kannappan/work/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:571:22)
    at IncomingMessage.onend (_stream_readable.js:660:10) {
  code: 'InvalidRequest',
  requestid: '0YAJM1FM2QTW719D',
  hostid: '8w89j+mgEKvwn7AIJl2GqCkGFuw+80C8JE5hQ6+bHmHSw34WDJIEjdPrQynIkGHwMPsc4qcA4jw=',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null
}
```

**Comment by kannappanr** [comment]
`content-md5` needs to be calculated for the body

**Comment by prakashsvmx** [comment]
@kannappanr  @ebozduman  Thank you for the review.

@kannappanr  I have updated and ran functional tests in standalone, distributed and gateway mode. 

Request you to PTAL.

**Comment by ebozduman** [comment]
There is a failure after fixing the missing md5 information in the request.
Here is my script:
```
var Minio = require('./dist/main/minio')
var s3Client = new Minio.Client({
  endPoint: 's3.amazonaws.com',
  // port: 9001,
  useSSL: true,
  region: 'us-east-1',
  accessKey: 'xxxxxxxxx',
  secretKey: 'xxxxxxxxxxxxxxxxx',
})
bucket = "retlock"
//Set Object lock on a bucket
s3Client.setObjectLockConfig(bucket, {mode:"COMPLIANCE",unit:'Days', validity:10}, function (err){
  if (err) {
    return console.log(err)
  }
  console.log("Success")
})

$ node run-set-object-lock-config.js
S3Error: The XML you provided was not well-formed or did not validate against our published schema
    at Object.parseError (/home/ersan/work/src/github.com/minio/minio-js/dist/main/xml-parsers.js:83:11)
    at /home/ersan/work/src/github.com/minio/minio-js/dist/main/transformers.js:158:22
    at DestroyableTransform._flush (/home/ersan/work/src/github.com/minio/minio-js/dist/main/transformers.js:82:10)
    at DestroyableTransform.prefinish (/home/ersan/work/src/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_transform.js:138:10)
    at DestroyableTransform.emit (events.js:315:20)
    at prefinish (/home/ersan/work/src/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:619:14)
    at finishMaybe (/home/ersan/work/src/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:627:5)
    at endWritable (/home/ersan/work/src/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at DestroyableTransform.Writable.end (/home/ersan/work/src/github.com/minio/minio-js/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at IncomingMessage.onend (_stream_readable.js:678:10) {
  code: 'MalformedXML',
  requestid: 'ZAMYAZKN34YER6FY',
  hostid: 'ZtH13VM6AzpL2oLHOKKaS0ZD2XL5ETbI8xtF/zsLH30I2NbRk1MCI6oJHoUiEEbBKGGTyayTNVM=',
  amzRequestid: null,
  amzId2: null,
  amzBucketRegion: null
}
```

**Comment by prakashsvmx** [comment]
@ebozduman @kannappanr  The issue was an extra key `Unit` was passed which the minio server ignored but it failed in S3. 
I have removed the same and tested against s3. It works fine now.

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**